[filebeat][ABS] - Segregated max_workers from batch/page size of the response (#44992)
elastic/beats: docs/reference/filebeat/filebeat-input-azure-blob-storage.md

5. [storage_url](#attrib-storage-url)
6. [containers](#attrib-containers)
7. [name](#attrib-container-name)
8. [batch_size](#attrib-batch_size-abs)
9. [max_workers](#attrib-max_workers)
10. [poll](#attrib-poll)
11. [poll_interval](#attrib-poll_interval)

> 13. [file_selectors](#attrib-file_selectors)
> 14. [expand_event_list_from_field](#attrib-expand_event_list_from_field)
> 15. [timestamp_epoch](#attrib-timestamp_epoch)
> 16. [custom_properties](#attrib-custom-properties)

Reviewer: The list _should_ still be rendered correctly the way you have it, but:

```suggestion
12. [file_selectors](#attrib-file_selectors)
13. [expand_event_list_from_field](#attrib-expand_event_list_from_field)
14. [timestamp_epoch](#attrib-timestamp_epoch)
15. [custom_properties](#attrib-custom-properties)
```

Tip: I usually just use `1.` in front of _all_ list items in ordered lists so if I have to reorder items or add/remove items I don't have to worry about it being out of order.

This is a specific subfield of a container. It specifies the container name.

## `batch_size` [attrib-batch_size-abs]

> This attribute specifies the `page size` for the response. In earlier versions, this value was derived from `max_workers`, but with the latest update, `batch_size` is now an independent setting. For backward compatibility, if `batch_size` is not explicitly defined, it will default to a value based on `max_workers`. This attribute can be configured at both the root and container levels. When defined at both levels, the container-level setting takes precedence.

Reviewer: Is `page size` an attribute? Should `page size` actually be `batch_size`? Or should it be `page_size` like `batch_size`? Or just "page size" (not formatted as code)?

Reviewer 2: I think this should be "page size" (not code formatted).

## `max_workers` [attrib-max_workers]

This attribute defines the maximum number of workers allocated to the worker pool for processing jobs which read file contents. It can be specified both at the root level of the configuration, and at the container level. Container level values override root level values if both are specified. Larger number of workers do not necessarily improve throughput, and this should be carefully tuned based on the number of files, the size of the files being processed and resources available. Increasing `max_workers` to very high values may cause resource utilization problems and may lead to bottlenecks in processing. Usually a maximum of `2000` workers is recommended. A very low `max_worker` count will drastically increase the number of network calls required to fetch the blobs, which may cause a bottleneck in processing.

::::{note}

> The `batch_size` and `max_workers` attributes are decoupled but functionally related. `batch_size` determines how many blobs are fetched in a single API call (i.e., the pagination size), while `max_workers` controls the number of concurrent goroutines used to process the fetched blobs. Although these values are independent, they should be configured thoughtfully to ensure efficient workload distribution and optimal performance. For example, setting `batch_size=100` and `max_workers=10` means each pagination request fetches `100` blobs, which are then processed by `10` concurrent goroutines. The appropriate value for `max_workers` depends on factors such as the number of files to be processed, available system resources, and network bandwidth.

Reviewer: We generally [avoid Latin abbreviations](https://eui.elastic.co/docs/content/word-choice/#latin-abbreviations).

```suggestion
The `batch_size` and `max_workers` attributes are decoupled but functionally related. `batch_size` determines how many blobs are fetched in a single API call (that is, the pagination size), while `max_workers` controls the number of concurrent goroutines used to process the fetched blobs. Although these values are independent, they should be configured thoughtfully to ensure efficient workload distribution and optimal performance. For example, setting `batch_size=100` and `max_workers=10` means each pagination request fetches `100` blobs, which are then processed by `10` concurrent goroutines. The appropriate value for `max_workers` depends on factors such as the number of files to be processed, available system resources, and network bandwidth.
```
